magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/ProbesService.scala

```diff
@@ -0,0 +1,62 @@
+package au.csiro.data61.magda.registry
+
+import java.util.concurrent.TimeoutException
+
+import akka.actor.{ActorRef, ActorSystem}
+import akka.event.Logging
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.Route
+import akka.stream.Materializer
+import au.csiro.data61.magda.client.AuthApiClient
+import au.csiro.data61.magda.directives.AuthDirectives.requireIsAdmin
+import au.csiro.data61.magda.directives.TenantDirectives.{
+  requiresTenantId,
+  requiresSpecifiedTenantId
+}
+import au.csiro.data61.magda.model.Registry._
+import com.typesafe.config.Config
+import gnieh.diffson.sprayJson._
+import io.swagger.annotations._
+import javax.ws.rs.Path
+import scalikejdbc.DB
+import org.everit.json.schema.ValidationException
+
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.concurrent.duration._
+import scala.util.{Failure, Success}
+
+// @Path("/status")
+class ProbesService(
+    config: Config,
+    authClient: RegistryAuthApiClient,
+    system: ActorSystem,
+    materializer: Materializer,
+    recordPersistence: RecordPersistence,
+    eventPersistence: EventPersistence
+) extends Protocols {
+  val logger = Logging(system, getClass)
+  implicit val ec: ExecutionContext = system.dispatcher
+
+  def live: Route = get {
+    path("live") {
+      complete("OK")
+    }
+  }
+
+  def ready: Route = get {
+    path("ready") {
+      DB readOnly { session =>
+        recordPersistence.getReadiness(Some(logger))(session)
+      } match {
+        case Some(true) => complete("OK")
+        case Some(false) | None =>
+          complete(StatusCodes.InternalServerError, "Database not ready")
+      }
+    }
+  }
```

Comment:
Tested in the cluster and it seems most exceptions are not captured.
e.g. 
```
java.util.NoSuchElementException: key not found: to_regclass
OR org.postgresql.util.PSQLException: Connection to registry-db:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
OR org.postgresql.util.PSQLException: FATAL: role "client" does not exist
```
I think you probably need to capture any failure from the DB.readonly method as the execution is happening there.
you might want to try scala's `util.Try` instead of `try/catch`  --- it's more common in scala as it's more flexible / offer more streamlined exception processing 
https://www.scala-lang.org/api/2.9.3/scala/util/Try.html 

You can do something like this:
```scala
import scala.util.{Try, Failure, Success}
Try{
        DB readOnly { session =>
          recordPersistence.getReadiness(Some(logger))(session)
        }
    } match {
        case Success(Some(true)) => complete("OK")
        case Success(Some(false)) | Success(None) =>
          complete(StatusCodes.InternalServerError, "Database not ready")
        case Failure(exception) => complete(StatusCodes.InternalServerError, "Database not ready Other")
    }
```
The `recordPersistence.getReadiness` doesn't seem to need to capture any exceptions (so you don't have to capture at multiple place) nor has a chance to produce a NONE. I guess a Boolean could simpler